tests/integration/test_charm.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
```

Comment:
We probably shouldn't be naming files "test_charm" anymore.
How about `test_charm_rules_reach_prometheus`?

Reply:
Why not?

Reply:
Let me elaborate on why I think this is okay. If I were to change the name, it would likely be to `test_basic_functionality.py` because that's sort of what is going on. If someone wants to write a test which checks that the scrape jobs get set up properly (and someone should), they should not have to rewrite `test_build_and_deploy`.

Reply:
If we can sensibly fit all itests in one file then `test_charm` makes sense.